gcloud/datastore/_implicit_environ.py

```diff
@@ -28,11 +28,20 @@
     app_identity = None
 
 
-DATASET_ID = None
-"""Module global to allow persistent implied dataset ID from enviroment."""
+class _DefaultsContainer(object):
+    """Container for defaults.
 
-CONNECTION = None
-"""Module global to allow persistent implied connection from enviroment."""
+    :type connection: :class:`gcloud.datastore.connection.Connection`
+    :param connection: Persistent implied connection from environment.
+
+    :type dataset_id: string
+    :param dataset_id: Persistent implied dataset ID from environment.
+    """
+
+    def __init__(self, connection=None, dataset_id=None, implicit=False):
```

Comment:
(re: missing comment:  I originally made it on the changeset, then moved it to the PR).